src/Converter/Feature2Texture.js

```diff
@@ -22,44 +23,29 @@ function drawPolygon(ctx, vertices, indices = [{ offset: 0, count: 1 }], style =
 
 function _drawPolygon(ctx, vertices, indices, style, size, extent, invCtxScale, canBeFilled) {
     // build contour
-    ctx.beginPath();
+    const path = new Path2D();
+
     for (const indice of indices) {
         if (indice.extent && Extent.intersectsExtent(indice.extent, extent)) {
             const offset = indice.offset * size;
             const count = offset + indice.count * size;
-            ctx.moveTo(vertices[offset], vertices[offset + 1]);
+            path.moveTo(vertices[offset], vertices[offset + 1]);
             for (let j = offset + size; j < count; j += size) {
-                ctx.lineTo(vertices[j], vertices[j + 1]);
+                path.lineTo(vertices[j], vertices[j + 1]);
             }
         }
     }
 
     // draw line or edge of polygon
     if (style.stroke) {
+        // TO DO add posibility of using a pattern and move to Style as fillPolygon()
         strokeStyle(style, ctx, invCtxScale);
-        ctx.stroke();
+        ctx.stroke(path);
     }
 
     // fill polygon only
     if (canBeFilled && style.fill) {
-        fillStyle(style, ctx, invCtxScale);
-        ctx.fill();
-    }
-}
-
-function fillStyle(style, ctx, invCtxScale) {
-    if (style.fill.pattern && ctx.fillStyle.src !== style.fill.pattern.src) {
-        ctx.fillStyle = ctx.createPattern(style.fill.pattern, 'repeat');
-        if (ctx.fillStyle.setTransform) {
-            ctx.fillStyle.setTransform(matrix.scale(invCtxScale));
-        } else {
-            console.warn('Raster pattern isn\'t completely supported on Ie and edge');
-        }
-    } else if (ctx.fillStyle !== style.fill.color) {
-        ctx.fillStyle = style.fill.color;
-    }
-    if (style.fill.opacity !== ctx.globalAlpha) {
-        ctx.globalAlpha = style.fill.opacity;
+        Style.prototype.applyToCanvasPolygon.call(style, ctx, matrix, invCtxScale, path);
```

Comment:
Because currently style is an object and not an instance of Style.
Style.drawingStylefromContext() return a style object and not a Style instance. This will be changed in the following PR.

Reply:
In a PR that is already created or that is to be created? If already created, which one ?

Reply:
In the 'parent' PR #2006 I merge drawingStyleFromContext() and symbolStylefromContext().
Both use styleOptions object and context to generate a Style combining all style informations. The idea is that this merged function, which can be rename getStyleFromContext(), should based on styleOptions objects and context instanciate a clear itowns Style Object (whith the default values and all).

Reply:
Ok great :+1: I resolve this conversation then